### PR TITLE
Portability fixes for preparation of NetBSD CI

### PR DESCRIPTION
**Pull request by tsutsui**
There are seveal Linux, Fedora, or bash specific features on build scripts and Makefile.

### CODE REVIEW

File: autogen.sh [hunk 1]
```diff
@@ -5,6 +5,9 @@
 : ${srcdir:=.}
 : ${SAVE_DIST_FILES:=0}
 
+if [ -z "$MAKE" ]; then
```
Comment (fujiwarat) [inline autogen.sh:8]: Why don't you use `: ${MAKE:=make}` ?

**Review by fujiwarat — COMMENTED** [review]
Please append `BUG=https://github.com/ibus/ibus/pull/2482` in the bug description.

File: autogen.sh (hunk 1 above)
Comment (tsutsui) [reply autogen.sh:8]: > Why don't you use `: ${MAKE:=make}` ?

Just I'm not familiar with make.

**Comment by tsutsui** [comment]
> Please append `BUG=https://github.com/ibus/ibus/pull/2482` in the bug description.

What do you mean 'the bug description'?
All commit messages?
Or should I squash all changes?

All changes are trivial but my fixes are tentative, so if you prefer proper implementation and commit history, I don't mind withdraw this pull request.

File: autogen.sh (hunk 1 above)
Comment (fujiwarat) [reply autogen.sh:8]: I ask to use `: ${MAKE:=make}` instead of `if [ -z "$MAKE" ] ; then`.

**Comment by fujiwarat** [comment]
> What do you mean 'the bug description'? All commit messages? Or should I squash all changes?

Sorry, I meant the patch description.
If you don't rebase your patch, it's fine to keep the current patch description and I will modify your patch later.

**Comment by tsutsui** [comment]
> > What do you mean 'the bug description'? All commit messages? Or should I squash all changes?
> 
> Sorry, I meant the patch description. If you don't rebase your patch, it's fine to keep the current patch description and I will modify your patch later.

I guess 'the patch description' means just 'commit messsage.'
If updated one (forced pushed one) doesn't match you intention, feel free to update changes.

**Pull request closed without merging**